windows/src/buildtools/mkver/Main.pas

```diff
@@ -34,8 +34,16 @@ implementation
   Xml.XMLIntf,
   Xml.XMLDoc;
 
+// This is the default build number used when doing a build outside of CI, and is
+// appended to the release version found in resources/version.md
+const
+  S_DebugBuildVersion = '9999';
+
+type
+  TMKVerMode = (mmUnknown, mmSetRootVersionFromBuild, mmSetRootVersionFromVersionMd, mmWriteVersionedFile, mmWriteManifestFile);
 var
-  IncrementVersion, DebugVersion: Boolean;
+  FMode: TMKVerMode = mmUnknown;
+  BuildVersion, ResourceMdFilename: string;
```

Comment:
I have not attempted to refactor mkver very much. It's very old and not great but there is not a lot of value to be gained in refactoring...